### PR TITLE
[Feature] State::load for verifreg and datacap.

### PR DESCRIPTION
**Pull request by ruseinov**
**Summary of changes**
Changes introduced in this pull request:
- State::load for verifreg and other tweaks to support getting data caps. 

**Reference issue to close (if applicable)**

work on https://github.com/ChainSafe/forest/pull/3817 

**Other information and links**

### CODE REVIEW

**Comment by ruseinov** [comment]
This has been investigated further, needs some changes to make it work properly!

**Comment by lemmih** [comment]
Let's copy over [get_balance](https://docs.rs/frc46_token/latest/src/frc46_token/token/state.rs.html#178-188) to `TokenState`. That should simplify the code a bit.

File: fil_actor_interface/src/builtin/datacap/mod.rs [hunk 1]
```diff
@@ -64,4 +73,52 @@ impl State {
...
+    pub fn verified_client_data_cap<BS>(
+        &self,
+        store: &BS,
+        addr: Address,
+    ) -> anyhow::Result<Option<BigInt>>
+    where
+        BS: Blockstore,
+    {
+        if addr.protocol() != Protocol::ID {
+            return Err(anyhow!("can only look up ID addresses"));
```
Comment (elmattic) [inline fil_actor_interface/src/builtin/datacap/mod.rs:86]: You can make use of `bail` macro here.

**Review by elmattic — APPROVED** [review]

**Comment by ruseinov** [comment]
> Let's copy over [get_balance](https://docs.rs/frc46_token/latest/src/frc46_token/token/state.rs.html#178-188) to `TokenState`. That should simplify the code a bit.

Sounds like a plan. Then we can make `hamt_bit_width` field private again. 

Hamt-related code is quite confusing, perhaps we could create an issue to review it and make more straightforward. For example, we're initialising a couple of maps in `TokenState`, but never use them. I understand that this is the code that was copied, but I think we should be keeping implementations minimal up until the point they need to be enriched.

**Comment by ruseinov** [comment]
> Let's copy over [get_balance](https://docs.rs/frc46_token/latest/src/frc46_token/token/state.rs.html#178-188) to `TokenState`. That should simplify the code a bit.

FYI: The `get_balance` does not line up well with Lotus implementation. I will modify it. We expect to return a `None` aka `null` when the balance is not found, but `get_balance` converts that into `0`.

**Review by aatifsyed — APPROVED** [review]

**Pull request merged**